pkg/cmd/validate.go

```diff
@@ -159,16 +161,33 @@ func (c *commandFlags) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	files, err := utils.FindFiles(args...)
-	if err != nil {
-		return err
+	var ioReaders []io.Reader
+	if args[0] != "-" {
```

Comment:
This forces `stdin` only to be available if it is the only "path" provided. Does this mirror `kubectl`'s behavior? What happens in kubectl if you provide `-` and a file?

Reply:
you mean by doing `-f - -f somefile.yaml`? I think they should both be read.

Reply:
And I agree, we should suppose `kubectl validate somefile.yaml - somedir/`

Reply:
For `kubectl` `-f - -f somefile.yaml`, seem to stuck reading stdin 
```
$ cat << EOF > pod.yaml
apiVersion: v1
kind: Pod
metadata:
  name: nginx
spec:
  containers:
  - name: nginx
    image: nginx:1.14.2
    ports:
    - containerPort: 80
EOF
$ kubectl apply -f - -f pod.yaml
```

Reply:
@alexzielenski , do you mean something like 
```
 cat pod.yaml | kubectl apply -f - -f pod1.yaml?
```

Reply:
> And I agree, we should suppose `kubectl validate somefile.yaml - somedir/`

Can the `-` works for a directory?

Reply:
Should the command work like? `-f` will give us a good idea of all the files given
```
kubectl validate -f file1 -f file2 -f - 
```

Reply:
> Can the - work for a directory?

What does it mean for stdin to be a directory? Not sure I understand that question.

> Should the command work like? `-f` will give us a good idea of all the files given

IMO, this was terrible on kubectl's end. I think they may require/ask for plugins to use that syntax, but that's always been pretty awful.

Reply:
> > What does it mean for stdin to be a directory? Not sure I understand that question.
> 
> the argument followed right after `-` should `stdin` right? but when you have `kubectl validate somefile.yaml - somedir/` , will it accept `somedir` as a valid argument since it's a directory instead of `stdin`?

there seems to be a misunderstanding. `-` means that it needs to read stdin on top of reading the other arguments given as a parameter. so basically, I would expect `kubectl validate somefile.yaml - somedir/` to validate the resources in somefile.yaml, those given through stdin, and all the resources recursively found in `somedir/`. One question might be, what do we do if we have multiple `-`? But I think repeated `-` can safely be ignored.

Reply:
Thanks for the clarification, appreciate it